### PR TITLE
3961 Hover item on hand switch

### LINKED ISSUE #3961: Item slot preview overlay doesn't update on switching hands.
## Description

Item slot preview overlay doesn't update on switching hands.

**Screenshots**

### PR DESCRIPTION
**Pull request by StrawberryMoses**
## About the PR 

Fixes #3961 

When switching hands, the client requests an update on the hover.

I do need some assistance on this, though- not sure if the update code is in the correct spot, because I get [WARN] net.ent: Got late MsgEntity! Diff: -3, msgT: 278, cT: 281, player: localhost@JoeGenero when the hands are switched and I'm not certain why.

**Screenshots**
handswitch

**Changelog**

:cl: Moses
- fix: Hand switching updates hover icon

### CODE REVIEW

**Comment by Zumorica** [comment]
When you open a PR to close a specific issue, please put it in the description like this: "Fixes #3961"
Putting the issue number in the title doesn't work

**Comment by StrawberryMoses** [comment]
> 
> 
> When you open a PR to close a specific issue, please put it in the description like this: "Fixes #3961"
> Putting the issue number in the title doesn't work

Literally did it right as you said it, totally forgot

**Comment by metalgearsloth** [comment]
You're calling SendHoverMessage when the game state is being rolled back which means the message is getting spammed over the network.

**Comment by StrawberryMoses** [comment]
What would be the appropriate place to call it?

**Comment by metalgearsloth** [comment]
> 
> 
> What would be the appropriate place to call it?

Somewhere that isn't being called during prediction. (beats me)

**Comment by mirrorcult** [comment]
i believe theres an event for hand switching and you can just subscribe to that?

File: Content.Server/Inventory/Components/InventoryComponent.cs [hunk 1]
```diff
@@ -275,7 +276,7 @@ public bool CanEquip(Slots slot, ItemComponent item, bool mobCheck, [NotNullWhen
                 reason = Loc.GetString("inventory-component-can-equip-cannot");
             }
 
-            var canEquip = pass && _slotContainers[slot].CanInsert(item.Owner);
+            var canEquip = pass && _slotContainers.ContainsKey(slot) && _slotContainers[slot].CanInsert(item.Owner);
```
[the remark this answers is not in the document]
Comment (StrawberryMoses) [reply Content.Server/Inventory/Components/InventoryComponent.cs]: Yeah I totally knew that 😎

File: Content.Server/Inventory/Components/InventoryComponent.cs [hunk 2]
```diff
@@ -608,5 +598,34 @@ public override bool IsEquipped(IEntity item)
...
+                Dirty();
+            }
+            else
+            {
+                _hoverEntity =
+                    new KeyValuePair<Slots, (EntityUid entity, bool fits)>(slot,
+                        (EntityUid.Invalid, false));
+
+                Dirty();
+            }
```
[the remark this answers is not in the document]
Comment (StrawberryMoses) [reply Content.Server/Inventory/Components/InventoryComponent.cs]: I replaced it with a TryGetComponent. 

(Was there something else icky here that I'm not seeing?)

**Comment by StrawberryMoses** [comment]
Yea, I dunno if there's anything client side that handles or predicts the hand switching? I didn't see anything, but I could've just missed something.

**Comment by metalgearsloth** [comment]
electro's predicting hands so that means this would need a complete rewrite to work with it and using the roundtrip was very ehh to begin with.

**Pull request closed without merging**